authservice/auth_service.go

```diff
@@ -26,10 +26,13 @@ type AuthService struct {
 	// ID of implementing service
 	serviceID string
 
-	services         *syncmap.Map
-	servicesUpdated  *time.Time
-	servicesLock     *sync.RWMutex
-	refreshCacheFreq int
+	services        *syncmap.Map
+	servicesUpdated *time.Time
+	servicesLock    *sync.RWMutex
+	HandlerMap      map[string]http.HandlerFunc
```

Comment:
Since we are pretty sure that the listener approach won't work, I think it would be best not to merge related code. Please move this code to a different branch so it will not be merged with this PR.